build.jam

```diff
@@ -0,0 +1,66 @@
+# Copyright René Ferdinand Rivera Morell 2023
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt)
+
+require-b2 5.2 ;
+
+project /boost/log
+    : common-requirements
+        <library>/boost/align//boost_align
+        <library>/boost/asio//boost_asio
+        <library>/boost/assert//boost_assert
+        <library>/boost/atomic//boost_atomic
+        <library>/boost/config//boost_config
+        <library>/boost/core//boost_core
+        <library>/boost/date_time//boost_date_time
+        <library>/boost/exception//boost_exception
+        <library>/boost/filesystem//boost_filesystem
+        <library>/boost/function_types//boost_function_types
+        <library>/boost/fusion//boost_fusion
+        <library>/boost/interprocess//boost_interprocess
+        <library>/boost/intrusive//boost_intrusive
+        <library>/boost/io//boost_io
+        <library>/boost/iterator//boost_iterator
+        <library>/boost/move//boost_move
+        <library>/boost/mpl//boost_mpl
+        <library>/boost/optional//boost_optional
+        <library>/boost/parameter//boost_parameter
+        <library>/boost/phoenix//boost_phoenix
+        <library>/boost/predef//boost_predef
+        <library>/boost/preprocessor//boost_preprocessor
+        <library>/boost/property_tree//boost_property_tree
+        <library>/boost/proto//boost_proto
+        <library>/boost/random//boost_random
+        <library>/boost/range//boost_range
+        <library>/boost/regex//boost_regex
+        <library>/boost/smart_ptr//boost_smart_ptr
+        <library>/boost/spirit//boost_spirit
+        <library>/boost/system//boost_system
+        <library>/boost/thread//boost_thread
+        <library>/boost/throw_exception//boost_throw_exception
+        <library>/boost/type_index//boost_type_index
+        <library>/boost/type_traits//boost_type_traits
+        <library>/boost/utility//boost_utility
+        <library>/boost/winapi//boost_winapi
+        <library>/boost/xpressive//boost_xpressive
+        <include>include
+    ;
+
+use-project /boost/log/atomic-int32 : config/atomic-int32 ;
+use-project /boost/log/message-compiler : config/message-compiler ;
+use-project /boost/log/native-syslog : config/native-syslog ;
+use-project /boost/log/pthread-mutex-robust : config/pthread-mutex-robust ;
+use-project /boost/log/regex-header-only : config/regex-header-only ;
+use-project /boost/log/x86-extensions : config/x86-ext ;
+use-project /boost/log/xopen-source-600 : config/xopen-source-600 ;
```

Comment:
Why are these here? Those are config checks needed for building the library. I would expect these projects to be referenced in `build/Jamfile.v2` or one of the `build/*-config.jam` files.